x/autopilot/keeper/ibc.go

```diff
@@ -0,0 +1,162 @@
+			return false, errorsmod.Wrapf(channeltypes.ErrInvalidAcknowledgement, "acknowledgement result cannot be empty")
+		}
+		return true, nil
+
+	case *channeltypes.Acknowledgement_Error:
+		k.Logger(ctx).Error(fmt.Sprintf("autopilot acknowledgement error: %s", response.Error))
+		return false, nil
+
+	default:
+		return false, errorsmod.Wrapf(channeltypes.ErrInvalidAcknowledgement, "unsupported acknowledgement response field type %T", response)
+	}
+}
+
+// Build an sdk.Coin type from the transfer metadata which includes strings for the amount and denom
+func (k Keeper) BuildCoinFromTransferMetadata(transferMetadata transfertypes.FungibleTokenPacketData) (coin sdk.Coin, err error) {
+	amount, ok := sdkmath.NewIntFromString(transferMetadata.Amount)
+	if !ok {
+		return coin, fmt.Errorf("unable to parse amount from transfer packet: %v", transferMetadata)
+	}
+	coin = sdk.NewCoin(transferMetadata.Denom, amount)
+	return coin, nil
+}
+
+// In the event of an ack error after a outbound transfer, we'll have to bank send to a fallback address
+func (k Keeper) SendToFallbackAddress(ctx sdk.Context, packetData []byte, fallbackAddress string) error {
+	// First unmarshal the transfer metadata to get the sender/reciever, and token amount/denom
+	var transferMetadata transfertypes.FungibleTokenPacketData
+	if err := transfertypes.ModuleCdc.UnmarshalJSON(packetData, &transferMetadata); err != nil {
+		return err
+	}
+
+	// Pull out the original sender of the transfer which will also be the bank sender
+	sender := transferMetadata.Sender
+	senderAccount, err := sdk.AccAddressFromBech32(sender)
+	if err != nil {
+		return errorsmod.Wrapf(err, "invalid sender address")
+	}
+	fallbackAccount, err := sdk.AccAddressFromBech32(fallbackAddress)
+	if err != nil {
+		return errorsmod.Wrapf(err, "invalid fallback address")
+	}
+
+	// Build the token from the transfer metadata
+	token, err := k.BuildCoinFromTransferMetadata(transferMetadata)
+	if err != nil {
+		return err
+	}
+
+	// Finally send to the fallback account
+	if err := k.bankKeeper.SendCoins(ctx, senderAccount, fallbackAccount, sdk.NewCoins(token)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// If there was a failed ack from an outbound transfer of one of the autopilot actions,
+// we'll need to check if there was a fallback address. If one was stored, bank send
+// to that fallback address
+// If the ack was successful, we should delete the address (if it exists)
+func (k Keeper) OnAcknowledgementPacket(ctx sdk.Context, packet channeltypes.Packet, acknowledgement []byte) error {
+	// Retrieve the fallback address for the given packet
+	// We use the packet source channel here since this will correspond with the channel on Stride
+	channelId := packet.SourceChannel
+	sequence := packet.Sequence
+	fallbackAddress, fallbackAddressFound := k.GetTransferFallbackAddress(ctx, channelId, sequence)
+
+	// If there was no fallback address, there's nothing else to do
+	if !fallbackAddressFound {
```

Comment:
1) In which scenarios could there be no fallback address? Is it just `this packet was not from an autopilot action` (if so, can we make sure we validate autopilot packets include a fallback address, otherwise we might traverse this codepath with an autopilot packet). Might help to add some of this context to the comment here. 
2) In this case, where will the funds go? I see we're returning `nil` error here so the IBC-transfer will be considered to have succeeded I believe

Reply:
Yeah so for context, the ack for every transfer will fall down this path. And the only transfer we have that has a fallback address (right now) is autopilot LS and forward.

I don't think we need to set a strict requirement that every autopilot packet has a fallback address, cause at the end of the day, it depends on the action.

But i think you can think of this function a bit more generally:
```
// function is intended to only handle fallback addresses
 
first, check did we register a fallback for this packet
      if no we didn't, then this function no longer has a purpose. pass the packet down to the transfer module to take over

however, if we did register a fallback, check if the transfer succeeded
       if it did, no need for a fallback since we only fallback in error cases
       if it failed, send funds to the fallback address
```

Reply:
> it depends on the action
> first, check did we register a fallback for this packet. if no we didn't, then this function no longer has a purpose. pass the packet down to the transfer module to take over

Ah these clear it up. Thanks. Resolving.